src/sentry/static/sentry/app/components/group/externalIssueActions.jsx

```diff
@@ -151,114 +153,189 @@ class ExternalIssueActions extends AsyncComponent {
   }
 }
 
-export class SentryAppExternalIssueActions extends React.Component {
-  static propTypes = {
-    group: PropTypes.object.isRequired,
-    sentryAppComponent: PropTypes.object.isRequired,
-    sentryAppInstallation: PropTypes.object,
-    externalIssue: PropTypes.object,
-  };
+export const SentryAppExternalIssueActions = withApi(
+  class SentryAppExternalIssueActions extends React.Component {
+    static propTypes = {
+      api: PropTypes.object.isRequired,
+      group: PropTypes.object.isRequired,
+      sentryAppComponent: PropTypes.object.isRequired,
+      sentryAppInstallation: PropTypes.object,
+      externalIssue: PropTypes.object,
+    };
 
-  constructor(props) {
-    super(props);
+    constructor(props) {
+      super(props);
 
-    this.state = {
-      action: 'create',
-      showModal: false,
+      this.state = {
+        action: 'create',
+        externalIssue: props.externalIssue,
+        showModal: false,
+      };
+    }
+
+    componentWillUpdate(prevProps) {
+      if (this.props.externalIssue !== prevProps.externalIssue) {
```

Comment:
Javascript object comparison is generally rubbish, do you want a deep equality here?

Reply:
Hmm, the intent here is essentially "did we not have an `externalIssue` before and now do" or vice versa. So mostly caring about a `null !== {}` comparison. It can't hurt to be more specific though.